content/docs/700-taming-complexity/index.md

```diff
@@ -0,0 +1,416 @@
+get in our way, introducing additional complexity that makes it hard for us to
+do what we want to do.
+
+<!-- If we can do Twitter embed, see @kentbeck's famous tweet on refactoring to make the hard thing easy -->
+
+Let's say we have a revolutionary new app idea: a todo list app. Our core
+application logic is simple -- it just fetches TODO items from a JSON API.
+
+```ts twoslash
+/// <reference path="node_modules/@types/node/index.d.ts" />
+// ---cut---
+interface TodoItem {}
+
+const getTodoPromise = async (id: number): Promise<unknown> => {
+  const res = await fetch(`https://jsonplaceholder.typicode.com/todos/${id}`)
+
+  return (await res.json()) as TodoItem
+}
+```
+
+See -- easy! But... this won't take us very far. There is a whole raft of "real world" complexity that we're not handling:
+
+- How can we efficiently get multiple todos at once?
+- What if the network goes down or the API misbehaves?
+- What if the user changes their mind after requesting a todo?
+- How can we test this code without hitting the real API?
+- What if the API requires authentication?
+- How do we manage the user's credentials responsibly?
+- How can we inspect our code's execution to be sure it's doing what we expect?
+
+These are all "incidental" complexity -- they feel like distractions from our
+application's true purpose. This is exactly the kind of complexity that Effect's
+abstractions are designed to help us solve; taming the complexity of the real
+world without getting in the way of our core application logic.
+
+## From Promises to Effects
+
+Before we get any further, we need to convert our promise-returning function
+into one that returns an Effect. The reasons for this will become clear as we
+use the Effect to manage the complexity above.
+
+```ts
+const getTodo = (id: number) =>
+  Effect.tryCatchPromise(
+    () => getTodoPromise(id),
+    (error) => new Error(`Uh-oh, something went wrong while getting todo #${id}!`),
+  )
+```
+
+We've wrapped our `getTodoPromise` function to make it return an `Effect`
+instead. What does that mean? The key is to look at the _type_ of `getTodo` (go
+ahead and hover your mouse over it above).
+
+As a [reminder](../200-what-is-effect/), an `Effect` type is composed of three
+things: the `R`equirements, the `E`rror, and the Success (`A`) types. By keeping
+the Error and Success types separate, we are documenting to consumers of
+`getTodo` that this function _should_ return a `TodoItem`, but it _may_ result
+in an error of type `Error` instead. We're also documenting that `getTodo` has
+no "Requirements", so our consumer can invoke the effect without providing
+anything more (we'll revisit that when we look at dependency injection later).
+
+Keep reading to find out how this simple abstraction can be powerful enough to
+tackle all the complexity described above!
+
+# Concurrency
+
+In the real world, we need to exploit concurrency. Fetching todos one-at-a-time
```

Comment:
may be worth to avoid "exploit" in favour of "leverage", I am personally not bothered but I know some interpret the terms in a non positive way